firebase/firestore.rules

```diff
@@ -1,8 +1,164 @@
+          ('description' in data) &&
+          ('latLng' in data) &&
+          ('status' in data) &&
+          ('createdAt' in data) &&
+          ('updatedAt' in data) &&
+          verifyHelpRequestAttributesForStatus(data.status, data)
+}
+
+function verifyOfferDataComplete(data) {
+   return ('cavUserRef' in data) &&
+          ('pinUserRef' in data) &&
+          ('requestRef' in data) &&
+          ('cavUserSnapshot' in data) &&
+          ('message' in data) &&
+          ('status' in data) &&
+          ('createdAt' in data)
+}
+
+// TODO: Further validate address object.
+function verifyUserPrivilegedInformationDataComplete(data) {
+   return ('address' in data) &&
+          ('termsAccepted' in data) &&
+          ('termsVersion' in data)
+}
+
 service cloud.firestore {
   match /databases/{database}/documents {
+    // Catch all -- Dont read/write to collections without rules
     match /{document=**} {
       allow read, write: if false;
     }
+
+    match /users/{userID} {
+      // Anyone can read your profile
+      allow read: if isAuthenticated();
+      // Only you can edit your own profile
+      allow write: if isUser(userID) && verifyUserDataComplete(request.resource.data);
+    }
+
+    // 1: Read + Writes to sub-collection can only be done by the owner
+    // 2: Writes to sub-collections must match userID chain so we can always directly reference
+    match /users/{userID}/privilegedInformation/{subDocID} {
+        allow read: if isUser(userID) && userID == subDocID;
+        allow write: if isUser(userID) && userID == subDocID && verifyUserPrivilegedInformationDataComplete(request.resource.data);
+    }
+
+    match /offers/{offerID} {
+      // Only the PIN or CAV of the request can read the offers
+      allow read: if (isVerifiedCAV() || isVerifiedPIN()) &&
+        confirmEitherPINorCAV(request.auth.uid, database, resource.data)
+      // Only CAVs can submit offers (as themselves)
+      allow write: if isVerifiedCAV() &&
+       currentUserIsCurrentCAVRef(request.auth.uid, database, request.resource.data) &&
+       verifyOfferDataComplete(request.resource.data);
+    }
+
+    match /requests/{reqID} {
+      // Only CAVs can see requests
+      allow read: if isVerifiedCAV() || isVerifiedPIN();
+      allow write: if isVerifiedPIN() &&
+       isNotSameCAVPIN(request.resource.data) &&
+       verifyHelpRequestDataComplete(request.resource.data);
+    }
+
+    match /questionnaires/{questID} {
+      // Can only read + write my own data
+      allow read: if isAuthenticated() && questionnaireBelongsToUser(request.auth.uid, database, resource.data);
+      // TODO: Change this rule once we allow other entities to submit questionnaires
+      allow write: if isAuthenticated() &&
+        verifyQuestionnaireComplete(request.resource.data) &&
+        questionnaireBelongsToUser(request.auth.uid, database, request.resource.data);
+    }
+
+    // Don't allow reading/writing to Teams and Orgs.
```

Comment:
How will those be managed?

Reply:
In the future, most likely a user will create them and become the `owner` which we will verify against over time.

Reply:
I should add this collection + model for Orgs and Teams is just a placeholder :)